packages/wdio-spec-reporter/src/index.js

```diff
@@ -387,18 +387,29 @@ class SpecReporter extends WDIOReporter {
     getEnviromentCombo (caps, verbose = true, isMultiremote = false) {
         const device = caps.deviceName
         const browser = isMultiremote ? 'MultiremoteBrowser' : (caps.browserName || caps.browser)
-        const version = caps.version || caps.platformVersion || caps.browser_version
-        const platform = caps.os ? (caps.os + ' ' + caps.os_version) : (caps.platform || caps.platformName)
+        /**
+         * fallback to different capability types:
+         * browserVersion: W3C format
+         * version: JSONWP format
+         * platformVersion: mobile format
+         * browser_version: invalid BS capability
+         */
+        const version = caps.version || caps.browserVersion || caps.platformVersion || caps.browser_version
+        /**
+         * fallback to different capability types:
+         * platformName: W3C format
+         * platform: JSONWP format
+         * os, os_version: invalid BS capability
+         */
+        const platform = caps.os ? (caps.os + ' ' + caps.osVersion) : caps.platformName || caps.platform
```

Comment:
I think the following order would be most logical for both values: w3c -> jswp -> mobile -> BS
What do you think @christian-bromann ?

Reply:
yeah, makes sense to me

Reply:
Great we finally agreed :)

Guys, may I ask you to commit `lint -fix` in this branch? Having trouble setting up it on Windows.

Reply:
What do you mean by that? Also I think what @erwinheitzman is suggestion is to change back the line 404 to:

```js
const platform = caps.platformName || caps.platform || caps.os ? (caps.os + ' ' + caps.osVersion) : '(unknown)'
```

Reply:
> Also I think what @erwinheitzman is suggestion is to change back the line 404 to:
> 
> ```js
> const platform = caps.platformName || caps.platform || caps.os ? (caps.os + ' ' + caps.osVersion) : '(unknown)'
> ```

like I said in https://github.com/webdriverio/webdriverio/pull/5390#discussion_r423650595 - it will not work

![image](https://user-images.githubusercontent.com/6696821/81691674-dab31a80-9465-11ea-80b1-766e829a4755.png)

That's the reason I updated it to 
```js
const platform = caps.os ? (caps.os + ' ' + caps.osVersion) : caps.platformName || caps.platform
```
@ericandrewlewis can you please clarify for us what you meant?

Reply:
@unickq can you please debug and see why this would return undefined? It seems strange that defining platform like this:
```js
const platform = caps.platformName || caps.platform || caps.os ? (caps.os + ' ' + caps.osVersion) : '(unknown)'
```
would return `undefined` given everything is concatenated with an `||`.

What @erwinheitzman meant was that we would prefer to pick valid standard based capabilities first before invalid ones.

Reply:
I debugged it  - ( ) missed
```js
const platform = caps.platformName || caps.platform || (caps.os ? caps.os + ' ' + caps.osVersion : '(unknown)')
```